pkgs/games/hmcl-bin/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, file
+, fetchurl
+, makeWrapper
+, jdk
+, alsaLib
+, xorg
+, wayland
+, libpulseaudio
+, libglvnd
+, autoPatchelfHook
+, libGL
+, glfw
+, openal }:
+
+stdenv.mkDerivation rec {
+  name = "hmcl-bin";
+  version = "3.5.3.223";
+
+  src = fetchurl {
+    url = "https://github.com/huanghongxun/HMCL/releases/download/v${version}/HMCL-${version}.jar";
+    sha256 = "sha256-8g2FMvAiAKfxJUY0G7wl6d44wOpVsknFHGZ85IkOzFc=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ jdk makeWrapper file ];
+  buildInputs = [ jdk ];
+
+  phases = [ "installPhase" ];
+  installPhase = let
```

Comment:
Missing runHook preInstall and runHook postInstall

Reply:
Fixed.